addons/project_hr_expense/models/project.py

```diff
@@ -16,14 +16,11 @@ def _compute_expenses_count(self):
         if not self.analytic_account_id:
             self.expenses_count = 0
             return
-        expenses_data = self.env['hr.expense']._read_group([
-            ('analytic_account_id', '!=', False),
-            ('analytic_account_id', 'in', self.analytic_account_id.ids)
-        ],
-        ['analytic_account_id'], ['analytic_account_id'])
-        mapped_data = {data['analytic_account_id'][0]: data['analytic_account_id_count'] for data in expenses_data}
         for project in self:
-            project.expenses_count = mapped_data.get(project.analytic_account_id.id, 0)
+            expenses = self.env['hr.expense'].search([
```

Comment:
This is introducing a database query in a loop. Can't we do that before the loop and use a dict in order to get the right value in the loop (same idea as what was done before)?

Reply:
It will be refactored in a near future by introducing indexes anyways. Not possible yet because we need JSON fields in the ORM but I was promised it would be done before the "real" freeze

Reply:
Ok I understand 😢
Will you take care of this when it will be available or should I schedule an activity in order to ensure that it gets improved?